test/integration/gardenlet/backupbucket/backupbucket_test.go

```diff
@@ -0,0 +1,285 @@
+				SecretRef: corev1.SecretReference{
+					Name:      secret.Name,
+					Namespace: secret.Namespace,
+				},
+				SeedName: &seed.Name,
+			},
+		}
+
+		Expect(testClient.Create(ctx, backupBucket)).To(Succeed())
+		log.Info("Created backupBucket for test", "backupBucket", client.ObjectKeyFromObject(backupBucket))
+
+		DeferCleanup(func() {
+			By("Delete BackupBucket")
+			Expect(testClient.Delete(ctx, backupBucket)).To(Or(Succeed(), BeNotFoundError()))
+		})
+
+		extensionSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateBackupBucketSecretName(backupBucket.Name),
+				Namespace: v1beta1constants.GardenNamespace,
+			},
+		}
+
+		extensionBackupBucket = &extensionsv1alpha1.BackupBucket{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: backupBucket.Name,
+			},
+		}
+
+		generatedSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateGeneratedBackupBucketSecretName(backupBucket.Name),
+				Namespace: seedGardenNamespace.Name,
+				Labels:    map[string]string{testID: testRunID},
+			},
+			Data: map[string][]byte{
+				"baz": []byte("dash"),
+			},
+		}
+
+		generatedSecretRef = &corev1.SecretReference{
+			Name:      generatedSecret.Name,
+			Namespace: generatedSecret.Namespace,
+		}
+
+		coreGeneratedSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateGeneratedBackupBucketSecretName(backupBucket.Name),
+				Namespace: gardenNamespace.Name,
+			},
+		}
+
+		coreGeneratedSecretRef = &corev1.SecretReference{
+			Name:      coreGeneratedSecret.Name,
+			Namespace: coreGeneratedSecret.Namespace,
+		}
+
+		expectedExtensionBackupBucketSpec = extensionsv1alpha1.BackupBucketSpec{
+			DefaultSpec: extensionsv1alpha1.DefaultSpec{
+				Type:           backupBucket.Spec.Provider.Type,
+				ProviderConfig: backupBucket.Spec.ProviderConfig,
+			},
+			Region: backupBucket.Spec.Provider.Region,
+			SecretRef: corev1.SecretReference{
+				Name:      extensionSecret.Name,
+				Namespace: extensionSecret.Namespace,
+			},
+		}
+
+		Expect(testClient.Create(ctx, generatedSecret)).To(Succeed())
+		log.Info("Created secret for test", "secret", client.ObjectKeyFromObject(generatedSecret))
+
+		DeferCleanup(func() {
+			By("Delete Secret")
+			Expect(testClient.Delete(ctx, generatedSecret)).To(Succeed())
+		})
+	})
+
+	Context("reconcile", func() {
+		BeforeEach(func() {
+			expectedExtensionBackupBucketSpec = extensionsv1alpha1.BackupBucketSpec{
+				DefaultSpec: extensionsv1alpha1.DefaultSpec{
+					Type:           backupBucket.Spec.Provider.Type,
+					ProviderConfig: backupBucket.Spec.ProviderConfig,
+				},
+				Region: backupBucket.Spec.Provider.Region,
+				SecretRef: corev1.SecretReference{
+					Name:      extensionSecret.Name,
+					Namespace: extensionSecret.Namespace,
+				},
+			}
+		})
+
+		It("should add the finalizer to the BackupBucket and the referred secret", func() {
+			Eventually(func(g Gomega) {
```

Comment:
You could use some `By("...")` statements to make the tests more verbose and easier readable (same below; check entire file and apply where appropriate)